ports/zephyr/machine_uart.c

```diff
@@ -84,20 +86,31 @@ STATIC mp_obj_t machine_uart_make_new(const mp_obj_type_t *type, size_t n_args,
 
     mp_map_t kw_args;
     mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
-    machine_uart_init_helper(self, n_args - 1, args + 1, &kw_args);
+    mp_machine_uart_init_helper(self, n_args - 1, args + 1, &kw_args);
 
     return MP_OBJ_FROM_PTR(self);
 }
 
-STATIC const mp_rom_map_elem_t machine_uart_locals_dict_table[] = {
-    { MP_ROM_QSTR(MP_QSTR_read), MP_ROM_PTR(&mp_stream_read_obj) },
-    { MP_ROM_QSTR(MP_QSTR_readline), MP_ROM_PTR(&mp_stream_unbuffered_readline_obj) },
-    { MP_ROM_QSTR(MP_QSTR_readinto), MP_ROM_PTR(&mp_stream_readinto_obj) },
-    { MP_ROM_QSTR(MP_QSTR_write), MP_ROM_PTR(&mp_stream_write_obj) },
-};
-STATIC MP_DEFINE_CONST_DICT(machine_uart_locals_dict, machine_uart_locals_dict_table);
+STATIC void mp_machine_uart_deinit(machine_uart_obj_t *self) {
+    (void)self;
+}
 
-STATIC mp_uint_t machine_uart_read(mp_obj_t self_in, void *buf_in, mp_uint_t size, int *errcode) {
+STATIC mp_int_t mp_machine_uart_any(machine_uart_obj_t *self) {
+    (void)self;
+    return 0; // TODO
+}
+
+STATIC bool mp_machine_uart_txdone(machine_uart_obj_t *self) {
+    (void)self;
+    return true;
```

Comment:
Suggest this either raises `NotImplementedError` or has a comment about why it's always true (does Zephyr TX never return before done?)

Reply:
This now raises.